scala/scala_cross_version.bzl

```diff
@@ -25,3 +25,109 @@ def scala_mvn_artifact(artifact):
   artifactid = gav[1]
   version = gav[2]
   return "%s:%s_%s:%s" % (groupid, artifactid, scala_version(), version)
+
+def _generate_scala_imports(version):
+  return """
+# scala.BUILD
+load("@io_bazel_rules_scala//scala:providers.bzl",
+     _declare_scalac_provider = "declare_scalac_provider",
+)
+
+java_import(
+    name = "scala-library",
+    jars = ["lib/scala-library.jar"],
+    visibility = ["//visibility:public"],
+)
+
+java_import(
+    name = "scala-compiler",
+    jars = ["lib/scala-compiler.jar"],
+    visibility = ["//visibility:public"],
+)
+
+java_import(
+    name = "scala-reflect",
+    jars = ["lib/scala-reflect.jar"],
+    visibility = ["//visibility:public"],
+)
+  """.format(version = version)
+
+def _generate_scalac_build_file_impl(repository_ctx):
+  scalac_worker_srcs = [
+      "CompileOptions.java",
+      "ScalaCInvoker.java",
+      "ScalacProcessor.java",
+      "Resource.java",
+  ]
+
+  for src in scalac_worker_srcs:
+    path = Label(
+        "@io_bazel_rules_scala//src/java/io/bazel/rulesscala/scalac:{}".format(
+            src))
+    repository_ctx.symlink(path, "scalac_worker_srcs_symlinked/{}".format(src))
```

Comment:
Good, I'd like to know if this is supported.

@ittaiz the reason it is not in the main workspace is to allow us to add multiple scala versions (as the scalac_worker needs to be compiled with a specific version of the compiler). So for this PR it could actually be in the main workspace (as we only use one scala version)

Reply:
In general I'll be very wary of creating symlinks pointing outside the external repo. Can this be a filegroup target instead?

Reply:
@dsmolov @ittaiz Filegroup doesn't work (or it works but logs warnings). The problem is that I want to compile a target in an external workspace with sources from the original workspace. This is required (or the only way I could come up with) if we want to have multiple dynamically defined scala versions in the same workspace.

If this is a no-go then I can update this PR to not do this (as this PR is only for choosing a single version), but this would make adding support for multiple versions harder.

Reply:
@ittaiz as I mentioned that gives me warnings: 
```
in srcs attribute of java_binary rule @io_bazel_rules_scala_scala_default//:scalac_worker: please do not import
'@io_bazel_rules_scala//src/java/io/bazel/rulesscala/scalac:CompileOptions.java' directly.
You should either move the file to this package or depend on an appropriate rule there
```

But if I don't dynamically generate a build file from a repository_rule as is currently done then I could use a filegroup. That would allow us to only have one scala version per workspace. For this PR I think I'll do that